internal/controller/metal3.io/baremetalhost_controller.go

```diff
@@ -1046,7 +1058,9 @@ func (r *BareMetalHostReconciler) actionInspecting(prov provisioner.Provisioner,
 	}
 
 	clearError(info.host)
-	info.host.Status.HardwareDetails = details
+	// To avoid further cluttering the BMH status with more information we'll remove the LLDP
+	// info before updating the status, but retain it for the HardwareData contents.
+	info.host.Status.HardwareDetails = removeLLDPInfo(details)
```

Comment:
This is a bit subjective for me, I would suggest not doing this operation. I understand that LLDP info will be duplicated but for me that is a smaller "issue" compared to doing extra CR modifications just to remove data that we have queried before. For me this looks like it is a extra work for not much functional benefit.

In order to try to keep the reconciliation as "simple and lightweight" I would not add this cleanup unless there is an actual functional problem caused by this few extra lines of inspection data on the BMH.

Reply:
I asked to do this because BMH will be unmanageable with LLDP data. I mean, it's already unmanageable... I don't feel strongly about this one but if we keep piling information into BMH, users will need to develop custom tools for handling BMH :D